### PR TITLE
Statistics refactoring

### PR DESCRIPTION
**Pull request by MykhailoKlapchuk**

### CODE REVIEW

**Review by bodnarira — APPROVED** [review]

**Review by MarianLabensky — APPROVED** [review]

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        /// <summary>
+        /// Method to get statistics for the cities for the years
+        /// </summary>
+        /// <param name="statisticsParameters">Statistics parameters: cities id's, years, indicators</param>
+        /// <returns>Statistics for the cities on the specified indicators and years</returns>
+        /// <response code="200">Successful operation</response>
+        /// <response code="403">User does not have access to the statistics</response>
+        /// <response code="404">The statistics does not exist</response>
+        [HttpGet("cities")]
+        public async Task<IActionResult> GetForCitiesForYears( [FromQuery] StatisticsParameters statisticsParameters)
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:41]: Could you explain name of this method ?

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            catch (NullReferenceException)
+            {
+                loggerService.LogError($"The statistics was not found");
+                return StatusCode(StatusCodes.Status404NotFound);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                loggerService.LogError($"User does not have access to the statistics");
+                return StatusCode(StatusCodes.Status403Forbidden);
+            }
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:58]: Why did you write this try/catch ?

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:89]: Why did you write this try/catch ?

File: EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs [hunk 4]
```diff
@@ -1,20 +1,49 @@
-﻿namespace EPlast.BLL.DTO.Statistics
+﻿using System.ComponentModel;
+
+namespace EPlast.BLL.DTO.Statistics
 {
     public enum StatisticsItemIndicator
     {
+        [Description("Кількість пташат")]
         NumberOfPtashata,
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs:9]: remove extra spaces

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs (hunk 1 above)
Comment (MykhailoKlapchuk) [reply EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:41]: I agree with you, it is weird name. Changed.

File: EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs (hunk 4 above)
Comment (MykhailoKlapchuk) [reply EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs:9]: I I disagree with you, it is more convenient to read with spaces.

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs (hunk 3 above)
Comment (MykhailoKlapchuk) [reply EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:89]: Changed

File: EPlast/EPlast.WebApi/Controllers/StatisticsController.cs (hunk 2 above)
Comment (MykhailoKlapchuk) [reply EPlast/EPlast.WebApi/Controllers/StatisticsController.cs:58]: Changed

**Review by LanchevychMaxym — APPROVED** [review]

File: EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs (hunk 4 above)
Comment (voroniak) [reply EPlast/EPlast.BLL/DTO/Statistics/StatisticsItemIndicator.cs:9]: Yes, but we do it either everywhere in the code or nowhere. Consistency must be maintained.

**Review by voroniak — APPROVED** [review]

**Pull request merged**